statix.lang/trans/generation/docs/modal.js

```diff
@@ -76,8 +74,20 @@ function btnClick(event) {
   for (const anchor of anchors) {
     anchor.addEventListener("click", closeModal);
   }
+  // Move the modal below the button
+  const nodeRect = node.getBoundingClientRect();
+  const marginLeft = nodeRect.left + window.scrollX;
+  const marginTop = nodeRect.bottom + window.scrollY;
+  modal.style.margin = `${marginTop + "px"} 0 0 ${marginLeft + "px"}`;
+  const pageLengthBefore = document.documentElement.scrollHeight;
   // Make the modal contents visible
   modal.style.display = "block";
+  // If the modal is outside the page, show it above the button
```

Comment:
Testing locally, this doesn't work for me: clicking on a button near the bottom of the screen, only part of the modal is visible:

![image](https://github.com/pdmosses/nabl/assets/18308236/36b396d4-0504-42a5-8c0c-e6feeecb4c14)

Reply:
Right, in the current implementation I only cared about the actual bottom of the page, not necessarily the viewport. I could change it to take into account the viewport.

Reply:
On a wide screen, centring the modal has the advantage of leaving the left side of the page content completely visible. Keywords and declared names tend to be on the left.

On a small screen (e.g., mobile) the modal is likely to take the whole width, and putting it near the top shows as much of it as possible.

So I'm wondering whether it might be better (and simpler) to revert to the original positioning?

Reply:
Reverted to original central positioning in the latest commits.